### ISSUE TITLE
Moving implementation details to private methods

### ISSUE THREAD

**Issue by ibnesayeed** [jekyll/classifier-reborn#126]
I am not pointing to anything specific, but perhaps we can revisit the API for `3.x` and move any implementation details to private methods. Currently only `LSI` class has three private methods and everything else is open.

**Comment by Ch4s3** [comment]
Yeah, let's start figuring out what needs to be private for 3.0, this will make documentation and maintenance a lot easier.
